### PR TITLE
UICIRCLOG-104: Circ Log - Implement MCL Next/Previous pagination

### PR DESCRIPTION
**Pull request by UladzislauKutarkin**
## Purpose

[UICIRCLOG-104](https://issues.folio.org/browse/UICIRCLOG-104)
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [x] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by UladzislauKutarkin** [comment]
@fe-tl-reviewers Please take a look at that PR. I can't request to review that team, so I'll try to do that through comments.

File: src/CirculationLogList/CirculationLogList.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ import {
   ResultsPane,
   NoResultsMessage,
   useLocationSorting,
-  useToggle,
+  useToggle, PrevNextPagination,
```
Comment (NikitaSedyx) [inline src/CirculationLogList/CirculationLogList.js:27]: move to next line plz  - better before hooks

File: src/CirculationLogList/CirculationLogListContainer.js [hunk 2]
```diff
@@ -10,15 +10,13 @@ import {
...
-  baseManifest,
-} from '@folio/stripes-acq-components';
+import { baseManifest, usePagination } from '@folio/stripes-acq-components';
 
 import { buildLogEventsQuery } from './utils';
 import { CirculationLogList } from './CirculationLogList';
+import { useCirculationLog } from '../hooks/useCirculationLog';
 
-const RESULT_COUNT_INCREMENT = 30;
+const RESULT_COUNT_INCREMENT = 50;
```
Comment (NikitaSedyx) [inline src/CirculationLogList/CirculationLogListContainer.js:19]: ticket says about 100 per page, 50 is acq specific

File: src/hooks/useCirlcuationLog.test.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    useLocation.mockReturnValue({ search: '?workflowStatus=Pending' });
+    rerender();
+
+    expect(result.current.isLoading).toBe(true);
+    await waitForNextUpdate();
+    expect(result.current.isLoading).toBe(false);
+  });
+
+  it('should call API right away and return records', async () => {
```
Comment (NikitaSedyx) [inline src/hooks/useCirlcuationLog.test.js:33]: this test is useless basically, it doesn't check what data is returned from hook

File: src/CirculationLogList/CirculationLogList.js (hunk 1 above)
Comment (UladzislauKutarkin) [reply src/CirculationLogList/CirculationLogList.js:27]: Thanks, I missed that

File: src/CirculationLogList/CirculationLogListContainer.js (hunk 2 above)
Comment (UladzislauKutarkin) [reply src/CirculationLogList/CirculationLogListContainer.js:19]: Already fixed!

**Review by vashjs — APPROVED** [review]

File: src/hooks/useCirlcuationLog.test.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import { useLocation } from 'react-router-dom';
+
+import { useCirculationLog } from './useCirculationLog';
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useLocation: jest.fn().mockReturnValue({}),
+}));
+
+describe('useList', () => {
```
[the remark this answers is not in the document]
Comment (UladzislauKutarkin) [reply src/hooks/useCirlcuationLog.test.js]: Also fixed that describe

File: src/hooks/useCirlcuationLog.test.js (hunk 3 above)
Comment (NikitaSedyx) [reply src/hooks/useCirlcuationLog.test.js:33]: this test still doesn't check records

File: src/CirculationLogList/CirculationLogList.js [hunk 5]
```diff
@@ -160,29 +162,43 @@ export const CirculationLogList = ({
...
+              onHeaderClick={changeSorting}
+              isEmptyMessage={resultsStatusMessage}
+              hasMargin
+              pagingType={null}
+              interactive={false}
+              height={height - PrevNextPagination.HEIGHT}
+              width={width}
+            />
+            {logEvents.length > 0 && (
+            <PrevNextPagination
```
Comment (NikitaSedyx) [inline src/CirculationLogList/CirculationLogList.js:193]: seems code style issue with codestyle, in other places I see indents after brackets

Comment (UladzislauKutarkin) [reply src/CirculationLogList/CirculationLogList.js:193]: Thanks for that, I will take a look

File: src/hooks/useCirlcuationLog.test.js (hunk 3 above)
Comment (UladzislauKutarkin) [reply src/hooks/useCirlcuationLog.test.js:33]: @NikitaSedyx , I made some investigation and found out that probably with the current implementation we are not able to check what returns from hook, coz we pass a function with a state that set that value of records to return.

Comment (NikitaSedyx) [reply src/hooks/useCirlcuationLog.test.js:33]: you can, as you have loadRecordsCB you can check how it's invoked + you can stub it in very simple way to just set records (set, records) => set(records), in this case you can check just records without checking callback call

**Review by NikitaSedyx — APPROVED** [review]

**Review by uladislausamets — APPROVED** [review]

**Pull request merged**